---
title: "Databases in annovarR"
author: "Jianfeng Li"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Databases in annovarR}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r, echo = FALSE, message = FALSE}
knitr::opts_chunk$set(comment = "#>", collapse = TRUE)
```

## Introduction

[BioInstaller](https://github.com/JhuangLab/BioInstaller) is a dependence project of [annovarR](https://github.com/JhuangLab/annovarR), which provides the download service of all annovarR supported databases.

```{r}
library(BioInstaller)
library(annovarR)
# Get all BioInstaller supported softwares, databases and files
BioInstaller::install.bioinfo(show.all.names = TRUE)
# Only db_annovar.toml in BioInstaller be included in annovarR
annovarR::download.database(show.all.names = TRUE)
```

In this reference manual, we arranged the databases in annovarR and provide several meta information about the annovarR supported databases and other BioInstaller supported (download only) databases. Some of description or comments have been described in [the download configuration file](https://github.com/JhuangLab/BioInstaller/blob/master/inst/extdata/config/db/db_annovar.toml) (BioInstaller package) and [the annotation configuration file](https://github.com/JhuangLab/annovarR/blob/master/inst/extdata/config/databases.toml) (annovarR).

## Databases

annovarR supported databases will firstly be supported to download from original sites (exclude the authentication part) in BioInstaller. A portion of BioInstaller supported databases will be introduced in annovarR to as the candidate annotation databases (process method: remain unchanged, re-formate, re-analysis).

### Gene and Clincal Annotation

Gene annotation databases contain the gene classification, gene function and phenotype correlation, such as [HGNC](https://www.genenames.org/), [OMIM](http://www.omim.org/) [DoCM](http://docm.genome.wustl.edu/), [CIVic](https://civic.genome.wustl.edu/), [DisGeNET](http://www.disgenet.org/web/DisGeNET/menu/browser?1), [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/), and Gene Ontology (GO), .etc.

### Variant Effect Prediction

Variant effect prediction databases contain the various databases generated by the algorithms for prediction of variants effect on protein or RNA structural, such as [SIFT](http://sift.jcvi.org/), [PolyPhen2](http://genetics.bwh.harvard.edu/pph2/), [PROVEAN](http://provean.jcvi.org), [MutationTaster](http://www.mutationtaster.org/), [MutationAssessor](http://mutationassessor.org/r3/), [FATHMM](http://fathmm.biocompute.org.uk/), .etc.

### Population Allele Frequency

Population allele frequency databases contain the databases based on the population cohort genome sequencing data (mainly include whole genome sequencing and whole exome sequencing), such as [1000 Genome Project](https://www.ncbi.nlm.nih.gov/projects/faspftp/1000genomes/), [NHLBI GO Exome Sequencing Project (ESP)](https://esp.gs.washington.edu/drupal/), [gnomAD](http://gnomad.broadinstitute.org/) and [ExAC](http://exac.broadinstitute.org/), .etc.

### Cancer Somatic Mutation

Cancer somatic mutation databases generated by the cancer patients case-control paired genomic sequence data, such as [COSMIC](http://cancer.sanger.ac.uk/cosmic), [Cancer Hotspots](http://cancerhotspots.org/#/home), [intogen](https://www.intogen.org/downloads) and [Cancer Biomarkers database](https://www.cancergenomeinterpreter.org/biomarkers), .etc.

### RNA-seq Variants

RNA-seq variants databases contributed by variants called from RNA-seq including expressed allele and RNA-editing. annovarR built an RNA-seq variants database, BRVar, based on 1285 cases B-cell lymphoblastic leukemia (B-ALL) patients RNA-seq data (Four different variants detection method be applied).

### Expression Quantitative Trait Locus (eQTL)

eQTL databases contain the candidate locus of genome that have an candidate impact on gene expression level, such as [Genotype-Tissue Expression (GTEx)](https://www.gtexportal.org/home/) QTL, [seeQTL](http://www.bios.unc.edu/research/genomic_software/seeQTL/) and [PancanQTL](http://bioinfo.life.hust.edu.cn/PancanQTL/), .etc.

### Non-coding RNA Related

Non-coding RNA databases contain the candidate biomaker or non-coding RNA targeted transcriptional regulation region, such as [Cancer-Specific CirRNA Database](http://gb.whu.edu.cn/CSCD/) and (LNCediting)[http://bioinfo.life.hust.edu.cn/LNCediting/], .etc.

**TODO: Finish full document in the next release.**
